docs/server/source/appendices/run-with-vagrant.md

```diff
@@ -0,0 +1,137 @@
+  the minimum versioning requirements.
+- Resources and specs for your box.
+  - RAM
+  - VCPUs
+  - Network Type
+    - Currently, only `private_network` is supported.
+  - IP Address
+- Setup type
+  - `quickstart`
+- Deploy node with Docker
+  - Deploy all the services in Docker containers or as processes.
+- Upstart Script
+- Vagrant Provider
+  - Virtualbox
+  - VMware
+
+## Install dependencies | Vagrant
+1. [VirtualBox](https://www.virtualbox.org/wiki/Downloads) >= 5.0.0
+2. [Vagrant](https://www.vagrantup.com/downloads.html) >= 1.16.0
+
+## Clone the BigchainDB repository | Vagrant
+```text
+$ git clone https://github.com/bigchaindb/bigchaindb.git
+```
+
+## Configuration | Vagrant
+Navigate to `bigchaindb/pkg/config/` inside the repository.
+```text
+$ cd bigchaindb/pkg/config/
+```
+
+Edit the `bdb-config.yaml` as per your requirements. Sample `bdb-config.yaml`:
+
+```text
+---
+- name: "bdb-node-01"
+  box:
+    name: "ubuntu/xenial64"
+  ram: "2048"
+  vcpus: "2"
+  setup_type: "quickstart"
+  deploy_docker: false
+  network:
+    ip: "10.20.30.40"
+    type: "private_network"
+  upstart: "/bigchaindb/scripts/bootstrap.sh"
+```
+
+**Note**: You can spawn multiple instances as well using `bdb-config.yaml`. Here is a sample `bdb-config.yaml`:
+```text
+---
+- name: "bdb-node-01"
+  box:
+    name: "ubuntu/xenial64"
+  ram: "2048"
+  vcpus: "2"
+  setup_type: "quickstart"
+  deploy_docker: false
+  network:
+    ip: "10.20.30.40"
+    type: "private_network"
+  upstart: "/bigchaindb/scripts/bootstrap.sh"
+- name: "bdb-node-02"
+  box:
+    name: "ubuntu/xenial64"
+  ram: "4096"
+  vcpus: "3"
+  setup_type: "quickstart"
+  deploy_docker: false
+  network:
+    ip: "10.20.30.50"
+    type: "private_network"
+  upstart: "/bigchaindb/scripts/bootstrap.sh"
+```
+
+
+## Local Setup | Vagrant
+To bring up the BigchainDB node, run the following command:
+
+```text
+$ vagrant up
+```
+
+*Note*: There are some vagrant plugins required for the installation, user will be prompted to install them if they are not present. Instructions to install the plugins can be extracted from the message.
+
+```text
+$ vagrant plugin install <plugin-name>
+```
+
+After successfull execution of Vagrant, you can log in to your fresh BigchainDB node.
+
+```text
+$ vagrant ssh <instance-name>
+```
+
+## Make your first transaction
+Once you are inside the BigchainDB node, you can verify that BigchainDB docker/process is running.
+
+Verify BigchainDB process:
+```text
+$ ps -ef | grep bigchaindb
+```
+
+OR
+
+Verify BigchainDB Docker:
+```text
+$ docker ps | grep bigchaindb
+```
+
+BigchainDB python driver is pre-installed in the instance, so you can instantly make transactions and verify the functionality. The `bdb_root_url` can be one of the following:
+```text
+# BigchainDB is running as a process
+bdb_root_url = http://<HOST-IP>:9984
+
+OR
+
+# BigchainDB is running inside a docker container
+bdb_root_url = http://<HOST-IP>:<DOCKER-PUBLISHED-HOST-PORT>
```

Comment:
Is the BigchainDB process inside the vagrant box run in a container?
The <OR> over here makes it confusing to know if it is a stand-alone process or a process inside a docker container.

Reply:
Yes, in `run-with-vagrant` guide it is running as a docker container inside the box.